questions/000-fn-pointer-vs-fn-type.md

```diff
@@ -0,0 +1,36 @@
+Answer: 4400
+Difficulty: 2
+
+# Hint
+
+The integer type parameters for `a`  have no affect in this example - they are a
+red herring.
+
+# Explanation
+
+There is no difference in size between any of `a::<u8>`,  `a::<u16>`, `a::<u32>`
+and `a::<u64>`. In fact all `fn` items have zero-sized types (ZST). To see why
+this makes sense, remember that the size of a closure is the combined size of
+its captured variables. `fn` items are not closures and cannot capture variables
+but they are interchangeable with non-capturing closures so it makes sense that
+their size should be zero.
+
+It is not possible to express the type of a specific function in Rust. This
+isn't usually a problem because we can rely on type inference and just constrain
+a type parameter with a member of one of the function trait families; `FnOnce`,
+`Fn` or `FnMut`.
+
+`fn` items implement these function traits, but Rust also provides a bit of
+magic for when you need to pass functions around but generics are inconvenient.
```

Comment:
I would object to describing any of the behavior as magical. I think it infantilizes the reader who is here because they want to understand Rust better. Glossing over something because it is magic comes across disrespectfully as "you would never understand this" or "you shouldn't want to understand this."

Reply:
I would say it's magical because it can't be explained in terms of other Rust features. I used the term because it was used in (perhaps older versions of) the Rust book for talking about things like how `Fn` traits are desugared.

Reply:
Makes sense, "you shouldn't want to understand this [right now]" is fine in the context of introductory material like the book. I think it would be better to phrase it some other way here.